src/weather_model_graphs/save.py

```diff
@@ -77,16 +86,20 @@ def to_pyg(
     def _get_edge_indecies(pyg_g):
         return pyg_g.edge_index
 
-    def _get_edge_features(pyg_g):
-        if edge_features != ["vdiff"]:
-            raise NotImplementedError(edge_features_values)
-        # TODO: handle features of different types more generally, i.e. both single ("len") values and tuples (like "vdiff")
-        return torch.cat((pyg_g.len.unsqueeze(1), pyg_g.vdiff), dim=1).to(torch.float32)
-
-    def _get_node_features(pyg_g):
-        if node_features != ["pos"]:
-            raise NotImplementedError(node_features_values)
-        return pyg_g.pos.to(torch.float32)
+    def _concat_pyg_features(pyg_g: "pyg.data.Data", features: List[str]) -> torch.Tensor:
+        """Convert features from pyg.Data object to torch.Tensor.
+        Each feature should be column in the resulting 2D tensor (n_features, n_edges or n_nodes).
+        Note, this function can handle node AND edge features.
+        """
+        v_concat = []
+        for f in features:
+            v = torch.tensor(pyg_g[f])  # make sure we have torch tensor
```

Comment:
Just for my understanding, why is this necessary? Do you have a specific case where `pyg_g[f]` is not tensor already?

Reply:
No, good point. The data from `pyg.data.Data` should always be tensors. I will remove it.